src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,47 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> top;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> next;
+
+        public Node(T value, Node<T> next) {
+            this.value = value;
+            this.next = next;
+        }
+    }
 
+    public void push(T value) {
+        Node<T> newNode = new Node<>(value, top);
+        top = newNode;
+        size++;
     }
 
     public T peek() {
-        return null;
+        isEmptyStack();
+        return top.value;
     }
 
     public T pop() {
-        return null;
+        isEmptyStack();
+        T oldValue = top.value;
```

Comment:
no need to call isEmtyStack() here and top.value;
you can just call oldValue = peek()
and as result you don't need method isEmptyStack(). you can check in peek

Reply:
Yes, that's true thanks.